src/Microsoft.ComponentDetection.Detectors/pip/SimplePythonResolver.cs

```diff
@@ -0,0 +1,374 @@
+    private SortedDictionary<string, IList<PythonProjectRelease>> ConvertSimplePypiProjectToSortedDictionary(SimplePypiProject simplePypiProject, PipDependencySpecification spec)
+    {
+        var sortedProjectVersions = new SortedDictionary<string, IList<PythonProjectRelease>>();
+        foreach (var file in simplePypiProject.Files)
+        {
+            try
+            {
+                var packageType = this.GetPackageType(file.FileName);
+                var version = this.GetVersionFromFileName(file.FileName);
+                var parsedVersion = PythonVersion.Create(version);
+                if (parsedVersion.Valid && parsedVersion.IsReleasedPackage &&
+                    PythonVersionUtilities.VersionValidForSpec(version, spec.DependencySpecifiers))
+                {
+                    var pythonProjectRelease = new PythonProjectRelease() { PythonVersion = version, PackageType = packageType, Size = file.Size, Url = file.Url };
+                    if (!sortedProjectVersions.ContainsKey(version))
+                    {
+                        sortedProjectVersions.Add(version, new List<PythonProjectRelease>());
+                    }
+
+                    sortedProjectVersions[version].Add(pythonProjectRelease);
+                }
+            }
+            catch (ArgumentException ae)
+            {
+                this.logger.LogError(
+                    ae,
+                    "Release {Release} could not be added to the sorted list of pip components for spec={SpecName}. Usually this happens with unexpected PyPi version formats (e.g. prerelease/dev versions).",
+                    JsonConvert.SerializeObject(file),
+                    spec.Name);
+                continue;
+            }
+        }
+
+        return sortedProjectVersions;
+    }
+
+    /// <summary>
+    /// Returns the package type based on the file name.
+    /// </summary>
+    /// <param name="fileName"> the name of the file from simple pypi. </param>
+    /// <returns>a string representing the package type.</returns>
+    private string GetPackageType(string fileName)
+    {
+        if (fileName.EndsWith(".whl"))
+        {
+            return "bdist_wheel";
+        }
+        else if (fileName.EndsWith(".tar.gz"))
+        {
+            return "sdist";
+        }
+        else if (fileName.EndsWith(".egg"))
+        {
+            return "bdist_egg";
+        }
+        else
+        {
+            return string.Empty;
+        }
+    }
+
+    /// <summary>
+    /// Uses regex to extract the version from the file name.
+    /// </summary>
+    /// <param name="fileName"> the name of the file from simple pypi. </param>
+    /// <returns> returns a string representing the release version. </returns>
+    private string GetVersionFromFileName(string fileName)
+    {
+        fileName = fileName.Replace(".tar", "-tar");
+        var version = Regex.Match(fileName, @"-(\d+\.((\d+)|(\d+\.(\d+))))-").Groups[1];
```

Comment:
I think we can get rid of the string replace and instead use this regex:

```regex
-(\d+\.\d+\.\d+)(.tar|-)
```

Reply:
Changed it to this `-(\d+\.\d+(\.\d)*)(.tar|-)` to match both versions like 1.0 and 1.0.0